src/System.Linq/src/System/Linq/First.cs

```diff
@@ -89,12 +89,40 @@ private static TSource TryGetFirst<TSource>(this IEnumerable<TSource> source, Fu
                 return ordered.TryGetFirst(predicate, out found);
             }
 
-            foreach (TSource element in source)
+            if (source is TSource[] array)
             {
-                if (predicate(element))
+                foreach (TSource element in array)
                 {
-                    found = true;
-                    return element;
+                    if (predicate(element))
+                    {
+                        found = true;
+                        return element;
+                    }
+                }
+            }
+            else if (source is List<TSource> list)
+            {
+                TSource element;
```

Comment:
Why is this declared outside of the loop?

Reply:
I believe I manually hoisted it out of the loop, but I guess it is safe to assume the compiler will do that for me. I have joined the declaration and assignment.